restli-docgen/src/main/java/com/linkedin/restli/docgen/DefaultDocumentationRequestHandler.java

```diff
@@ -260,5 +269,5 @@ private static RoutingException createRoutingError(String path)
   private RestLiDocumentationRenderer _jsonRenderer;
   private RestLiConfig _config;
   private Map<String, ResourceModel> _rootResources;
-  private final AtomicBoolean _shouldInitialize = new AtomicBoolean(true);
+  private boolean _initialized = false;
```

Comment:
Wondering if `volatile` keyword is needed for the variable?

Reply:
May not be needed since they use a double check pattern where _initialized is rechecked/manipulated within a synchronized block.